src/components/ViewChannelItem.jsx

```diff
@@ -399,6 +412,20 @@ function ViewChannelItem({ channelObjectProp, loadTeaser, playTeaser, minimal, p
     setToolTipheight(containerHeight?.top);
   };
 
+  const onAllowModalOpen = () => {
+    if (checkPermission() === NOTIF_PERMISSION_STATUS.PENDING) {
```

Comment:
```suggestion
    if (checkPermission() === "pending") {
```

Reply:
should we check this using an enum or string . i think enum will be better so that whenever there is any change we can directly change the enum at one place @rohitmalhotra1420

Reply:
We are not checking it with it a string only. Because we are declaring a return type for checkPermission function, vs code will automatically read the output of this function and will give us and error if try to cmpare that output with some random string.

eg.

```
const checkPermission=():NotificationPermissionType=> {
    return "pending" or "granted"
}

if(checkPermission==="granted"){} // this is correct.

if(checkPermission==="accepted"){} // typescript automatically detects it with intellisense and gives error.
```

Reply:
If we use enum we have to import those enums everywhere to check for conditions.

Reply:
understood